csrc/multidevice/utils.cpp

```diff
@@ -408,6 +409,30 @@ bool haveDifferentShardings(
         return false;
       }
 
+      if (consumer->definition()->isA<ViewOp>()) {
+        // Consider the reshape: [h]-> [a, h/a] where `h` and `a` are sharded,
+        // IdModel currently does not map the two DID axes. Similarly for the
+        // corresponding merge reshape. This is a temporary workaround to check
+        // if the reshape is resharding.
+        NVF_ERROR(
```

Comment:
This is incorrect since we do not have this split if the DID parallelization is on logical domain.

Reply:
(TODO) I need to add 2 things here:
1. Check if the sharding is on the reshaped IDs for both producer and consumer, else use IdModel.
2. If the sharding is on reshaped IDs, it is outer split and on the outer reshaped ID.